scan/tls_session.go

```diff
@@ -0,0 +1,87 @@
+package scan
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+)
+
+// TLSSession contains tests of host TLS Session Resumption via
+// Session Tickets and Session IDs
+var TLSSession = &Family{
+	Name:        "TLSSessionScanners",
+	Description: "Scans host's implementation of TLS session resumption using session tickets/session IDs",
+	Scanners: []*Scanner{
+		{
+			"SessionResumeScanner",
+			"Host is able to resume sessions accross all addresses.",
+			sessionResumeScan,
+		},
+	},
+}
+
+// sessionCache is a mock ClientSessionCache used to record all stores (i.e. new session tickets)
+type sessionCache struct {
```

Comment:
I've actually removed this type entirely and used `tls.NewLRUClientSessionCache()` instead